### PR TITLE
Fixed 5 flaky tests

### PR DESCRIPTION
**Pull request by alany9552**
## Description
Fixed 5 flaky tests causes by non-deterministic results using `HashMap`

### Flaky tests and root causes
Folllowing tests are flaky and failing:
```
org.apache.iotdb.tsfile.read.controller.IMetadataQuerierByFileImplTest.testConvert1
org.apache.iotdb.tsfile.read.controller.IMetadataQuerierByFileImplTest.testConvert2
org.apache.iotdb.tsfile.read.ReadInPartitionTest.test1
org.apache.iotdb.tsfile.read.ReadInPartitionTest.test2
org.apache.iotdb.tsfile.read.ReadInPartitionTest.test3
```
when running
`mvn -pl cluster edu.illinois:nondex-maven-plugin:1.1.2:nondex -Dtest=*TESTNAME*`

The reason is because it uses `HashMap` when declaring `groupWriters` which is non-deterministic and mess up the orders. 
### How to fix and results
Using LinkedHashMap will solve this problem and produce deterministic results which won't fail the test.
After changing, all tests pass.

### Error Log
```
[ERROR] Failures:
[ERROR]   IMetadataQuerierByFileImplTest.testConvert2:148 expected:<[[( 148056261804]0 : 1480562618045 ]]> but was:<[[[ 148056261800]0 : 1480562618045 ]]>
```

<hr>

This PR has:
- [X] been self-reviewed.
    - [X] concurrent read
    - [X] concurrent write
    - [X] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

##### Key changed/added classes (or packages if there are too many classes) in this PR
**org.apache.iotdb.tsfile.read.controller.IMetadataQuerierByFileImplTest**

### CODE REVIEW

**Comment by jixuan1989** [comment]
Hi @alany9552 , can you fix the conflicts of this PR? (you can omit CI failure of E2E Tests now).

**Comment by alany9552** [comment]
> Hi @alany9552 , can you fix the conflicts of this PR? (you can omit CI failure of E2E Tests now).

Sure I will try to work on this.

**Comment by alany9552** [comment]
> Hi @alany9552 , can you fix the conflicts of this PR? (you can omit CI failure of E2E Tests now).

I have already fixed the conflict.

**Comment by alany9552** [comment]
Hi @jixuan1989 I have already fixed the conflicts, could you please check my PR?

**Comment by HTHou** [comment]
Hi, Plz Run 'mvn spotless:apply' to fix the format violations.

**Comment by alany9552** [comment]
@HTHou Hi, thanks for reminding. I have fixed the format violations.

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/TsFileWriter.java [hunk 1]
```diff
@@ -45,8 +45,8 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
```
Comment (HTHou) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/write/TsFileWriter.java:49]:
```suggestion
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
```

**Pull request closed without merging**